### DISCUSSION TITLE
Nunit TestCaseSource, how do i avoid Guids

### DISCUSSION THREAD

**Discussion by infinitejest69** [VerifyTests/Verify#1363] [Q&A]
Say i have an APi test like this

```
        [TestCaseSource(nameof(ByIDCases))]
        public async Task BranchesByID(Guid someGuid)
        {
                //Act
                var result = await client.GetAsync(someGuid);
                //Assert
                await Verify(result, _verifySettings);
        }
 ```
 
and my source source looking like this

```
        private static IEnumerable<TestCaseData> ByIDCases()
        {
            yield return new TestCaseData(Guid.NewGuid()).SetName("GetByID Unkown Guid"); 
            yield return new TestCaseData(Guid.Empty).SetName("GetByID Empty Guid")                ;
        }
```
Q:
1. can i use the test name in the SetName as a parameter for the verify file?
2. if not how can avoid the Guid as a parameter

**Comment by SimonCropp** [comment]
i was not aware of the SetName API. i will release a new version that adds explicit support

**Reply by infinitejest69** [reply]
Thanks for this, should i just watch out for this fix on the Verify Repo
assuming it will make it into this read me

https://github.com/VerifyTests/Verify/blob/main/docs/parameterised-nunit.md

**Comment by infinitejest69** [comment]
OK thanks i can now do what i need but not quite the way i expected, leaving this for future reference

the problem above was not solved using unknown parameter [types](https://github.com/VerifyTests/Verify/blob/main/docs/parameterised-nunit.md#unknown-parameter-types) as i am verifying the result the Guid was still used in the name but setting the [override](https://github.com/VerifyTests/Verify/blob/main/docs/parameterised-nunit.md#overriding-text-used-for-parameters)

to solve adding the setting in the OneTimeSetup
```

            _verifySettings.UseTextForParameters(TestContext.CurrentContext.Test.Name);
```
